src/MudBlazor/Base/MudBaseButton.cs

```diff
@@ -128,7 +127,8 @@ protected virtual async Task OnClickHandler(MouseEventArgs ev)
             if (GetDisabledState())
                 return;
             await OnClick.InvokeAsync(ev);
-            Activatable?.Activate(this, ev);
+            if (!Class?.Contains("mud-input-adornment-icon-button") ?? true)
```

Comment:
Does this tweak work? Might be easier to parse

```suggestion
            if (Class?.Contains("mud-input-adornment-icon-button") != true)
```

It doesn't seem right to add special conditions in the base button itself, but I don't fully get the IActivatable stuff. Would like to know what @ScarletKuro thinks

Reply:
I was able to move the activator logic to MudMenu without much fanfare.

Reply:
> It doesn't seem right to add special conditions in the base button itself

I don't like it either, even if it's in MudMenu now. But I guess there is no other good ways? How we will deal with it when we will continue with removing `IActivatable`?